cpp/src/IceLocatorDiscovery/PluginI.cpp

```diff
@@ -599,44 +589,30 @@ LocatorI::foundLocator(const Ice::LocatorPrxPtr& locator)
         }
     }
 
-    Ice::LocatorPrxPtr l = _pendingRequests.empty() ? _locators[locator->ice_getIdentity().category] : _locator;
-    if(l)
+    auto i = _locators.find(locator->ice_getIdentity().category);
+    if (i != _locators.end())
     {
-        //
-        // We found another locator replica, append its endpoints to the
-        // current locator proxy endpoints.
-        //
-        Ice::EndpointSeq newEndpoints = l->ice_getEndpoints();
+        // We found another locator replica, append its endpoints to the current locator proxy endpoints.
+        Ice::EndpointSeq newEndpoints = i->second->ice_getEndpoints();
         Ice::EndpointSeq endpts = locator->ice_getEndpoints();
-        for(Ice::EndpointSeq::const_iterator p = endpts.begin(); p != endpts.end(); ++p)
+        for (const auto& endpoint : locator->ice_getEndpoints())
         {
-            //
-            // Only add endpoints if not already in the locator proxy endpoints
-            //
-            bool found = false;
-            for(Ice::EndpointSeq::const_iterator q = newEndpoints.begin(); q != newEndpoints.end(); ++q)
-            {
-                if(*p == *q)
-                {
-                    found = true;
-                    break;
-                }
-            }
-            if(!found)
+            if (std::find(newEndpoints.begin(), newEndpoints.end(), endpoint) == newEndpoints.end())
             {
-                newEndpoints.push_back(*p);
+                newEndpoints.push_back(endpoint);
             }
         }
-        l = l->ice_endpoints(newEndpoints);
+        i->second = i->second->ice_endpoints(newEndpoints);
     }
     else
     {
-        l = locator;
+        _locators.insert(make_pair(locator->ice_getIdentity().category, locator));
     }
 
-    if(_pendingRequests.empty())
+    Ice::LocatorPrx l = i == _locators.end() ? locator : i->second;
+
+    if (_pendingRequests.empty())
     {
-        _locators[locator->ice_getIdentity().category] = l;
```

Comment:
Not clear why we have this before. The code above just insert the locator in the locators map or updates its endpoints if it was already there.